crates/typst/src/layout/align.rs

```diff
@@ -442,6 +450,14 @@ impl Add<HAlignment> for VAlignment {
     }
 }
 
+impl Resolve for VAlignment {
+    type Output = FixedAlignment;
+
+    fn resolve(self, styles: StyleChain) -> Self::Output {
+        self.fix(TextElem::dir_in(styles))
```

Comment:
No need to retrieve the text direction if it isn't used. If `VAlignment` really needs to implement `FixAlignment`, we could just pass `Dir::TTB`.